cmake/Modules/FindDPDK.cmake

```diff
@@ -0,0 +1,181 @@
+# ~~~
+# - Try to find DPDK include dirs and libraries
+#
+# Find the DPDK includes and client library
+# This module defines:
+#  DPDK_FOUND, If false, do not try to use DPDK.
+#  DPDK_INCLUDE_DIRS, where to find rte_config.h and rte_version.h
+#  DPDK_LIBRARIES, the libraries needed by a DPDK user
+#  DPDK_CFLAGS_OTHER, the compile flags to use
+#  DPDK_VERSION, the version of the library
+# ~~~
+
+include(FindPackageHandleStandardArgs)
+
+function(DPDK_READ_VERSION DPDK_VERSION DPDK_VERSION_FILE)
+  if(NOT DPDK_VERSION_FILE)
+    return()
+  endif()
+
+  file(READ "${DPDK_VERSION_FILE}" DPDK_VERSION_STR)
+  string(
+    REGEX MATCH
+          "#define RTE_VER_YEAR ([0-9]+)"
+          _
+          ${DPDK_VERSION_STR})
+  set(DPDK_VERSION_MAJOR ${CMAKE_MATCH_1})
+
+  string(
+    REGEX MATCH
+          "#define RTE_VER_MONTH ([0-9]+)"
+          _
+          ${DPDK_VERSION_STR})
+  set(DPDK_VERSION_MINOR ${CMAKE_MATCH_1})
+
+  string(
+    REGEX MATCH
+          "#define RTE_VER_MINOR ([0-9]+)"
+          _
+          ${DPDK_VERSION_STR})
+  set(DPDK_VERSION_PATCH ${CMAKE_MATCH_1})
+
+  set(DPDK_VERSION
+      "${DPDK_VERSION_MAJOR}.${DPDK_VERSION_MINOR}.${DPDK_VERSION_PATCH}"
+      PARENT_SCOPE)
+endfunction()
+
+# Try to find DPDK with pkg-config first!
+find_package(PkgConfig QUIET)
+if(PKG_CONFIG_FOUND)
+  pkg_check_modules(
+    DPDK
+    QUIET
+    IMPORTED_TARGET
+    libdpdk>=18.11)
+endif()
+
+# We found using Pkg-Config!
+if(DPDK_FOUND)
+  message("DPDK found with pkg-config")
+  if(DPDK_DEBUG)
+    message("-----------")
+    message("Libraries: ${DPDK_LIBRARIES}")
+    message("Link Libraries: ${DPDK_LINK_LIBRARIES}")
+    message("Library DIr: ${DPDK_LIBRARY_DIRS}")
+    message("Ldflags: ${DPDK_LDFLAGS}")
+    message("Include Dirs: ${DPDK_INCLUDE_DIRS}")
+    message("Cflags: ${DPDK_CFLAGS}")
+    message("Cflags Other: ${DPDK_CFLAGS_OTHER}")
+    message("Version: ${DPDK_VERSION}")
+    message("-----------")
+  endif()
+else()
+  message("DPDK not found with pkg-config trying legacy mode")
+
+  # Find the include dirs and get the version
+  find_path(DPDK_VERSION_INCLUDE_DIR rte_version.h REQUIRED PATH_SUFFIXES dpdk include)
+  find_path(DPDK_CONFIG_INCLUDE_DIR rte_config.h REQUIRED PATH_SUFFIXES dpdk include)
+
+  set(DPDK_INCLUDE_DIRS ${DPDK_INCLUDE_DIRS})
+  list(APPEND DPDK_INCLUDE_DIRS ${DPDK_VERSION_INCLUDE_DIR})
+  list(APPEND DPDK_INCLUDE_DIRS ${DPDK_CONFIG_INCLUDE_DIR})
+  list(REMOVE_DUPLICATES DPDK_INCLUDE_DIRS)
+
+  dpdk_read_version(DPDK_VERSION "${DPDK_VERSION_INCLUDE_DIR}/rte_version.h")
```

Comment:
In newer versions of DPDK the version info is actually in `rte_build_config.h `, not in `rte_version.h`. Maybe we should look at both of them starting with  `rte_build_config.h `?

Reply:
Do you have a docker with this version just to test ?

Reply:
Yes, here is a docker image you can test with: `seladb/ubuntu2004-dpdk2011`

Reply:
this should be good with latest version

Reply:
should be good now